internal/bgp/bgp.go

```diff
@@ -671,3 +686,28 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the address addr exists on any of the
+// network interfaces in the ifs slice. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(ifs []net.Interface, addr net.IP) bool {
+	for _, i := range ifs {
+		addresses, err := i.Addrs()
+		if err != nil {
```

Comment:
Just curious: when can this happen? If the interface doesn't have any address, does this return an error?

I feel that can be a frequent case (like interfaces in a bond, or some unused vti, etc.). But that is guessing when that fails, maybe it doesn't fail in that case?

If it does, I think it will be fine to change the `return false` to just `continue`. In that way, we ignore that interface and if none matches we return false at the end of the function anyways.

Reply:
`Addrs()` returns an error so we want to do the error checking here. I'm not sure exactly when an error is returned.

Thanks, I actually changed `return false` to `continue` and apparently undid that change :shrug: Fixing.

Reply:
Same goes for the 2nd `return false` - if one address is broken, maybe the next one is OK.